grails-app/services/org/pih/warehouse/core/ILocationRoleDataService.groovy

```diff
@@ -0,0 +1,9 @@
+package org.pih.warehouse.core
+
+interface ILocationRoleDataService {
```

Comment:
Let's leave the interface as LocationRoleDataService and use something different for the implementation class (BaseLocationRoleDataService, AbstractLocationRoleDataService, etc).

Using the I-interface-name convention feels very Microsoft / C# to me. I have nothing against Hungarians but the Hungarian notation should have been burned with fire long ago.

Some light reading 
https://developer.okta.com/blog/2019/06/25/iinterface-considered-harmful
https://stackoverflow.com/questions/2814805/java-interfaces-implementation-naming-convention
https://stackoverflow.com/questions/541912/interface-naming-in-java
https://softwareengineering.stackexchange.com/questions/117348/should-interface-names-begin-with-an-i-prefix

Reply:
Actually, you don't need the interface at all. Just throw all of the interface methods into the class as abstract methods like this
https://github.com/MTUHIDE/CCLC-Queue/blob/535d339ebce367440f129e69efd460c96836e8da/grails-app/services/edu/mtu/cs/queue/QueueAnswerService.groovy#L5

Reply:
If you did have an interface and an implementation class, you could also put them in the same file which feels like a better convention to me.

https://github.com/rundeck/rundeck/blob/ee6fedc65d917ebb910ab0787c85e95a95ff8b6a/rundeckapp/grails-app/services/rundeck/services/data/ScheduledExecutionDataService.groovy#L19

Reply:
@jmiranda to be honest... I also hate the `ILocationDataService` naming, but the reason I went that path since in GORM data services docs they tend to use it.

Reply:
@jmiranda I've moved the methods to the abstract class as you suggested